crates/solidity/outputs/npm/package/src/generated/index.d.ts

```diff
@@ -648,20 +649,17 @@ export enum ProductionKind {
 export class RuleNode {
   get type(): NodeType.Rule;
   get kind(): RuleKind;
-  get byteRange(): [start: number, end: number];
-  get charRange(): [start: number, end: number];
-  get byteRangeIncludingTrivia(): [start: number, end: number];
-  get charRangeIncludingTrivia(): [start: number, end: number];
+  get utf8TextLen(): number[];
```

Comment:
I think the return type regressed here? Is this intentional?

Reply:
No, it wasn't :)

Reply:
For the `[utf8: number, utf16: number, char: number]` type, I wonder at what time is it better to use an object `TextIndex` with named properties (`utf8`/`utf16`/`char`) over the tuple. Is it cheap to serialize the object to TS side at once, given that we don't need to use identity or send it back to rust.

Not blocking for this PR though.

Reply:
I had the same thought, particularly after adjust the TS test and seeing that I had to use array indices.

Reply:
#509